packages/git-encrypted/src/encryptedMeta.ts

```diff
@@ -0,0 +1,96 @@
+import zod from 'zod';
+import { superpathjoin as join } from 'superpathjoin';
+
+import { FS, GitBaseParamsEncrypted } from './types';
+import { generateKey } from './crypto';
+import { encryptedRepoCommit } from './git';
+
+const META_FILENAME = 'encrypted.json';
+
+const EncryptedMetaSchema = zod.object({
+  version: zod.literal(1),
+  derivationParams: zod.object({
+    salt: zod.string().nonempty(),
+    cpuCost: zod
+      .number()
+      .int()
+      .positive(),
+    blockSize: zod
+      .number()
+      .int()
+      .positive(),
+    parallelizationCost: zod
+      .number()
+      .int()
+      .positive(),
+  }),
+});
+type EncryptedMeta = zod.infer<typeof EncryptedMetaSchema>;
+
+const createEncryptedMeta = (): EncryptedMeta => {
+  return {
+    version: 1,
+    derivationParams: {
+      salt: generateKey(1024),
```

Comment:
nit: I would prefer to have `const salt = generateKey()` as it is clearer, and makes it easier to see where errors arise, etc.

Do we want a 1024 byte salt? That seems extremely long to me, but I'm not sure what makes sense here. I'd assume more like 32 or so, but I could be wrong.

Also, I guess it makes sense to make this value a constant and put it, probably at the top of this file. It's not really a global constant that we need to share elsewhere in the codebase I guess.

Reply:
I agree with salt constant nit.

About the length, my understanding is that the longer the better, and we only use that function once per repository creation, so it shouldn't be a problem performance-wise.

I appreciate the question, not sure about that.